hazelcast/test/src/sql_test.cpp

```diff
@@ -505,6 +866,392 @@ TEST_F(SqlTest, statement_with_params)
+                       "FROM %1% "
+                       "WHERE height = CAST(? AS REAL) OR age = ?") %
+         map_name)
+          .str()
+    };
+
+    auto height_param = expecteds.front().height;
+    auto age_param = expecteds.back().age;
+
+    statement.set_parameters(height_param);
+    statement.clear_parameters();
+    statement.set_parameters(height_param, age_param);
+
+    ASSERT_NO_THROW(client.get_sql().execute(statement).get());
+}
+
+TEST_F(SqlTest, test_execute_statement_with_mismatched_params_when_sql_has_more)
+{
+    create_mapping();
+    (void)populate_map();
+
+    sql::sql_statement statement{
+        client,
+        (boost::format("SELECT * FROM %1% WHERE __key > ? AND this > ?") %
+         map_name)
+          .str()
+    };
+
+    statement.set_parameters(5);
+    auto execution = client.get_sql().execute(statement);
+
+    EXPECT_THROW(execution.get(),
+                 hazelcast::client::sql::hazelcast_sql_exception);
+}
+
+TEST_F(SqlTest,
+       test_execute_statement_with_mismatched_params_when_params_has_more)
+{
+    create_mapping();
+    (void)populate_map();
+
+    sql::sql_statement statement{
+        client,
+        (boost::format("SELECT * FROM %1% WHERE this > ?") % map_name).str()
+    };
+
+    statement.set_parameters(5, 6);
+    auto execution = client.get_sql().execute(statement);
+
+    EXPECT_THROW(execution.get(), sql::hazelcast_sql_exception);
+}
+
+TEST_F(SqlTest, test_execute_statement_with_timeout)
+{
+    create_mapping_for_student();
+    auto expecteds = populate_map<test::student>(100);
+
+    sql::sql_statement statement{
+        client,
+        (boost::format("SELECT __key, height FROM %1% WHERE height < 100") %
+         map_name)
+          .str()
+    };
+
+    statement.timeout(std::chrono::milliseconds{ 100 });
+
+    auto expected_row_count = count_if(
+      begin(expecteds), end(expecteds), [](const test::student& student) {
+          return student.height < 100.0f;
+      });
+
+    auto result = client.get_sql().execute(statement).get();
+
+    for_each_row(result,
+                 assert_row_count(expected_row_count),
+                 [&expecteds](const sql::sql_page::sql_row& row) {
+                     auto idx = row.get_object<int>(0);
+
+                     EXPECT_TRUE(idx.has_value());
+                     auto height = row.get_object<float>(1);
+                     EXPECT_TRUE(height.has_value());
+                     EXPECT_EQ(expecteds.at(*idx).height, *height);
+                 });
+}
+
+TEST_F(SqlTest, test_statement_with_wrong_timeout)
+{
+    sql::sql_statement statement{ client, "non_null_query" };
+
+    ASSERT_THROW(statement.timeout(std::chrono::milliseconds{ -500 }),
+                 exception::illegal_argument);
+}
+
+TEST_F(SqlTest, test_statement_with_empty_sql)
+{
+    auto fn = [this]() { sql::sql_statement statement{ client, "" }; };
+
+    ASSERT_THROW(fn(), exception::illegal_argument);
```

Comment:
```suggestion
    ASSERT_THROW(sql::sql_statement{ client, "" }, exception::illegal_argument);
```

and delete the other line.

Reply:
`,` is interpreted as argument seperator this is why I moved it to the lambda function.